src/commands/php_list.rs

```diff
@@ -0,0 +1,95 @@
+use clap::App;
+use clap::SubCommand;
+use std::env;
+use std::path::Path;
+use std::path::PathBuf;
+use std::str;
+use glob::glob;
+use std::process::Command;
+use std::process::Stdio;
+use regex::Regex;
+
+pub(crate) fn command_config<'a, 'b>() -> App<'a, 'b> {
+    SubCommand::with_name("php:list").about("List all available PHP executables")
+}
+
+pub(crate) fn php_list() {
+    let path_string = env::var_os("PATH").unwrap();
+    let path_dirs = path_string
+        .to_str()
+        .unwrap()
+        .split(get_path_separator().as_str())
+        .collect::<Vec<&str>>()
+    ;
+
+    let mut binaries: Vec<String> = vec![];
+
+    for dir in &path_dirs {
+        let path = Path::new(dir);
+        binaries.append(&mut binaries_from_path(path.to_owned()));
+    }
```

Comment:
First, you could optimise the vector allocation such as: `Vec::with_capacity(path_dirs.len())`, so that you avoid re-allocation if the vector is too small.

Second, there is a much better way to do this!

```suggestion
    let binaries: Vec<String> = path_dirs
         // consumes `path_dirs` and produce an iterator
        .into_iter()
        .map(|dir| {
            // use `PathBuf` directly instead of `Path::new(…).into_owned()`
            binaries_from_path(PathBuf::from(dir))
        })
        // at that point, `binaries_from_path` returns a `Vec<String>` so the iterator is of kind “`Vec<Vec<String>>`”, we only want `Vec<String>`
        .flatten()
        // enjoy!
        .collect();
```